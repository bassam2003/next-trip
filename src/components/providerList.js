import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import { Paper, Button, Menu, MenuItem } from '@material-ui/core'

const styles = {
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    margin: '20px',
  },
  button: {
    margin: '20px',
  }
}
export class ProviderList extends React.PureComponent {
  state = {
    anchorEl: null,
  }

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget })
  }

  handleClose = () => {
    this.setState({ anchorEl: null })
  }

  handleSelectAndClose = (currentProvider) => {
    const { onSelectProvider = () => {} } = this.props
    onSelectProvider(currentProvider)
    this.setState({ anchorEl: null })
  }

  render() {
    const { providers = [], currentProvider = '', classes = {} } = this.props
    const { anchorEl } = this.state
    const currentProviderData = providers.find(x =>  x.Value ===  currentProvider) || {}
    return (
      <Paper className={classes.root}>
        <h4>Filter By Provider: </h4>
        <Button
          aria-owns={anchorEl ? 'simple-menu' : undefined}
          aria-haspopup="true"
          onClick={this.handleClick}
          className={classes.button}
        >
          {currentProviderData.Text || "All Providers" }
        </Button>
        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          <MenuItem 
              onClick={() => this.handleSelectAndClose('')}
            >
              All Providers
            </MenuItem>
          {providers.map((provider, idx) => 
            <MenuItem 
              key={idx}
              onClick={() => this.handleSelectAndClose(provider.Value)}
            >
              {provider.Text}
            </MenuItem>
          )}
        </Menu>
      </Paper>
    )
  }
}

export default withStyles(styles)(ProviderList)