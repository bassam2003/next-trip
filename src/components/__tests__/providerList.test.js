import React from 'react'
import { shallow } from 'enzyme'
import { ProviderList } from '../providerList'

describe('ProviderList', () => {
  it('should render without error', () => {
    const wrapper = shallow(<ProviderList/>)
    expect(wrapper).toMatchSnapshot()
  })
})