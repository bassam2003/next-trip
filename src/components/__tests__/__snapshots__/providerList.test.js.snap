// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderList should render without error 1`] = `ShallowWrapper {}`;
